x-pack/legacy/plugins/uptime/public/components/functional/location_map/location_status_tags.tsx

```diff
@@ -20,54 +27,97 @@ const BadgeItem = styled.div`
 
 const TagContainer = styled.div`
   padding: 10px;
-  max-height: 200px;
+  max-height: 229px;
   overflow: hidden;
 `;
 
+const OtherDiv = styled.div`
+  padding-left: 18px;
+`;
+
 interface Props {
   locations: MonitorLocation[];
 }
 
+interface StatusTag {
+  label: string;
+  timestamp: number;
+}
+
 export const LocationStatusTags = ({ locations }: Props) => {
   const {
     colors: { gray, danger },
   } = useContext(UptimeSettingsContext);
 
-  const upLocs: string[] = [];
-  const downLocs: string[] = [];
+  const upLocations: StatusTag[] = [];
+  const downLocations: StatusTag[] = [];
 
   locations.forEach((item: any) => {
     if (item.summary.down === 0) {
-      upLocs.push(item.geo.name);
+      upLocations.push({ label: item.geo.name, timestamp: new Date(item.timestamp).valueOf() });
     } else {
-      downLocs.push(item.geo.name);
+      downLocations.push({ label: item.geo.name, timestamp: new Date(item.timestamp).valueOf() });
     }
   });
 
+  // Sort by recent timestamp
+  upLocations.sort((a, b) => {
+    return a.timestamp < b.timestamp ? 1 : b.timestamp < a.timestamp ? -1 : 0;
+  });
+
+  moment.locale('en', {
+    relativeTime: {
```

Comment:
I'd envision this set of values like this:

```typescript
  moment.locale('en', {
    relativeTime: {
      future: 'in %s',
      past: '%s ago',
      s: '%ds',
      ss: '%ss',
      m: '%dm',
      mm: '%dm',
      h: '%dh',
      hh: '%dh',
      d: '$dd',
      dd: '%dd',
      M: '%d Mon',
      MM: '%d Mon',
      y: '%d Yr',
      yy: '%d Yr',
    },
  });
```

This results in an output like:
`3s ago`
`2m from now`
`3h ago`
`1 Mon ago`
`2 Yr ago`

In most cases, years and months should not factor into the UI, so I'm more ok with longer names that will help delineate those units, especially in cases where it could be a month, because it will be very easy to confuse something like `1m` and `1M`.

Interested in additional feedback from @andrewvc @katrin-freihofner @shahzad31 @drewpost

Reply:
Looks good to me. Thinking through it, maybe we can even drop `ago` to save some space. Is it obvious to others that this is past tense?

Also, how often do these values update out of curiosity? If someone disable refresh I feel like these values should still recalculate.

Reply:
> Thinking through it, maybe we can even drop ago to save some space. Is it obvious to others that this is past tense?

I initially suggested we could drop `ago` as well to save real estate. The drawback is if something weird did happen (I'm thinking if they adjust the date range or something if the like) we could end up with strange values. I think we could leave the `in %s` for the future and it would clarify if somehow the values being rendered had weirdness.

> Also, how often do these values update out of curiosity? If someone disable refresh I feel like these values should still recalculate.

These refresh along with the app's refresh. If someone disables `auto refresh` then they will stop rendering. I think that making it behave independently would add some challenge because it also follows the established refresh interval, so if I change it from the default 60-second span to, say, 6 seconds, the values will update and mirror that. Adding that complexity may make it less feasible to hit the upcoming release, but we should probably focus on getting this to work in what we determine to be the right way, IMO.

Reply:
I have to admit that I would prefer `min` instead of `m` but I double-checked and we use `m` in other areas as well. 👍

Reply:
i am also inclined towards using 'min' and 'sec' , i believe there is enough space to cater for those values.

Reply:
Though i agree about other changes.

Reply:
I'm ok with `min` and `sec` if there's a consensus. If we agree about removing relative qualifiers (`ago`, `from`, etc.) I don't think the space savings for the unit names is imperative.

Reply:
++. As far as the refreshing values outside of autorefresh, we could add that in a subsequent PRs.

Reply:
About refreshing data, i think, it's an interesting idea, we should have some sort of Callout for whole application, i have seen some sites, "You are looking at outdated, refresh to get updated data." In case of auto refresh is disabled." We can do something like that. Might be hard to decide on the threshold.

Reply:
That solution sounds good Shahzad - I think the threshold should be whatever our default refresh interval is. So we'd use `60s` today.